test/jdk/java/lang/invoke/TestCatchExceptionWithVarargs.java

```diff
@@ -26,7 +26,10 @@
  * @bug 8019184
  * @library /test/lib /java/lang/invoke/common
  * @summary MethodHandles.catchException() fails when methods have 8 args + varargs
- * @run main TestCatchExceptionWithVarargs
+ * @comment The maximum node limit below avoids, in combination with -Xcomp,
+ *          timeouts due to compilation of a large number of methods with a
+ *          large number of parameters.
+ * @run main/othervm -XX:MaxNodeLimit=15000 TestCatchExceptionWithVarargs
```

Comment:
Why not have two runs here. One that requires that there is no `Xcomp`, where we have the normal node limit. And one where you lower the node limit, so that `Xcomp` is ok?

Reply:
Same here, I just added the `MaxNodeLimit`. I'd prefer leaving any other changes to a separate RFE (if needed).